torch/csrc/distributed/c10d/init.cpp

```diff
@@ -190,18 +219,24 @@ They are used in specifying strategies for reduction collectives, e.g.,
 
           .def(
               "allreduce",
-              &::c10d::ProcessGroup::allreduce,
+              [](::c10d::ProcessGroup& pg,
+                 std::vector<at::Tensor> xs,
+                 ::c10d::AllreduceOptions opts) {
+                xs = unbox(xs);
+                return pg.allreduce(xs, opts);
+              },
               py::arg("tensors"),
               py::arg("opts") = ::c10d::AllreduceOptions(),
               py::call_guard<py::gil_scoped_release>())
 
           .def(
               "allreduce",
               [](::c10d::ProcessGroup& pg,
-                 std::vector<at::Tensor>& xs,
+                 std::vector<at::Tensor> xs,
```

Comment:
It turns out pybind always converts Python containers to STL containers in place. Because of that, the pass by reference is useless. I only found out when working on this change, so changed it here. I'll change it back for this commit and update every call site in this file to take arguments by value in one big sweeping change instead of only changing these ones.